Feat/static generation

jimhlad opened the pull request:
This is a modified version of @pi0 's PR for static image generation (took the heavy lifting from there).

Would be interested to get a second pair of eyes from @danielroe or someone on the Nuxt team.

P.S. Not sure if creating a separate PR was the best approach. Feel free to close this out if it's way off the mark or if modifying the original PR is easier. Thanks!

jimhlad commented:
Updated to address issue outlined here: https://github.com/nuxt/image/pull/614#issuecomment-1299377666

jimhlad commented:
Updated to fix issue with srcset image generation

husayt commented:
This works great. Please merge asap

brunodeangelis commented:
Given that #614 has seen some activity today by @pi0 , which one should be merged?

pi0 commented on src/runtime/ipx.ts:
```diff
@@ -1,16 +1,10 @@
-  const ipxOptions = {
-    ...(useRuntimeConfig().ipx || {}),
-    // TODO: Switch to storage API when ipx supports it
-    dir: fileURLToPath(new URL('../public', import.meta.url))
```
This workaround works also for the production `.output/` directory structure  as path is resolved at runtime. If we do `resolve()` in `src/ipx.ts`, full path of build environment will be into the bundle.

pi0 commented:
Thanks for your work on this @jimhlad please keep doing this ❤️  For now, I went with #614 solution.

jimhlad commented:
Thanks @pi0 .

Does it work with multiple formatters at the same time though (e.g. width, height **and** quality)?

I recall I had to add `trim()` here to fix something (I forget what it was though):

```
...(sizes.value.srcset || '').split(',').map(s => s.trim().split(' ')[0])
```

Also had some issues using `process` in the `<nuxt-img>` component. But maybe fixed now 😄

pi0 commented:
Feel free to open PR for other issues ❤️
